modules/microshift-lvm-thin-volumes.adoc

```diff
@@ -0,0 +1,47 @@
+// Module included in the following assemblies:
+//
+// * microshift_storage/volume-snapshots-microshift.adoc
+
+:_content-type: CONCEPT
+[id="microshift-lvm-thin-volumes_{context}"]
+= About LVM thin volumes
+
+To use advanced storage features such as creating volume snapshots or volume cloning, you must provision a logical volume manager (LVM) thin-pool on the {op-system-ostree} host and configure both the logical volume manager storage (LVMS) provider and the cluster. LVM thin-pools must be attached to a volume group.
+
+[IMPORTANT]
+====
+To create CSI snapshots, you must configure thin volumes on the {op-system-ostree} host. The CSI does not support volume shrinking.
+====
+
+For LVMS to manage thin logical volumes (LVs), a thin-pool `device-class` must be specified in the `etc/lvmd.yaml` configuration file. Multiple thin-pool device classes are permitted.
+
+If additional storage pools are configured with device classes, then additional storage classes must also exist to expose the storage pools to users and workloads. Specifically, to enable such dynamic provisioning on a thin-pool, a `StorageClass` must be present on the cluster that specifies the source `device-class` in the `topolvm.io/device-class` parameter.
+
+.Example `lvmd.yaml` that specifies a single device class for a thin-pool
+
+[source, yaml]
+----
+socket-name: <1>
+device-classes: <2>
+  - name: thin <3>
+    default: true
+    spare-gb: 0 <4>
+    thin-pool:
+      name: thin
+      overprovision-ratio: 10 <5>
+    type: thin <6>
+    volume-group: ssd <7>
+----
+[.small]
+<1> String. The UNIX domain socket endpoint of gRPC. Defaults to  '/run/lvmd/lvmd.socket'.
+<2> A list of maps for the settings for each `device-class`.
+<3> String. The unique name of the `device-class`.
+<4> Unsigned 64-bit integer. Storage capacity in GB to be left unallocated in the volume group. Defaults to `0`.
+<5> If you have multiple thin-provisioned devices that share the same pool, then these devices can be over-provisioned. Over-provisioning requires a float value of 1 or greater.
+<6> Thin provisioning is required to create volume snapshots.
+<7> String. The group where the `device-class` creates the logical volumes.
+
+[IMPORTANT]
+====
+A race condition prevents LVMS from accurately tracking the allocated space and preserving the `spare-gb` for a device class when multiple PVCs are created simultaneously. Use separate volume groups and device classes to protect the storage of highly dynamic workloads from each other.
```

Comment:
`spare-gb` needs to be followed by a noun, but I think it doesn't make sense if you were to say "preserving the `spare-bg` field for a a device class..." Maybe use "storage capacity" from callout `<4>` instead?

Reply:
Consider starting the sentence with the "when" clause, if that maintains the right meaning

```suggestion
When multiple PVCs are created simultaneously, a race condition prevents LVMS from accurately tracking the allocated space and preserving the `spare-gb` for a device class. Use separate volume groups and device classes to protect the storage of highly dynamic workloads from each other.
```